src/App.tsx

```diff
@@ -34,62 +35,111 @@ export function getReorderedGoods(
-  console.log(sortType, isReversed);
+  visibleGoods.sort((f1, f2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return f1.localeCompare(f2);
+
+      case SortType.LENGTH:
+        return f1.length - f2.length;
+
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean,
+  sortType: SortType,
+}
+
+export class App extends React.PureComponent<{}, State> {
+  state: Readonly<State> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  handleSortByAlphabet = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  handleSortByLenght = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames('button is-info', {
+              'is-light': sortType !== SortType.ALPHABET,
+            })}
+            onClick={this.handleSortByAlphabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-success', {
+              'is-light': sortType !== SortType.LENGTH,
+            })}
+            onClick={this.handleSortByLenght}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames('button is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={this.handleReverse}
+          >
+            Reverse
+          </button>
+          {(isReversed || sortType !== SortType.NONE) && (
```

Comment:
move to the separate variable
```suggestion
          {... && (
```